airflow/contrib/kubernetes/pod_launcher.py

```diff
@@ -96,9 +98,14 @@ def run_pod(self, pod, startup_timeout=120, get_logs=True):
 
         return self._monitor_pod(pod, get_logs)
 
+    @tenacity.retry(
```

Comment:
I'm not sure we want to retry here -- most of the individual components have retries on, and if it fails after we've called read_pod_logs then all the container logs would be duplicated.

Reply:
The other methods have retries with `stop_after_attempt(3)`. This is not sufficient for long running tasks, because during periods of unstable network conditions they have a high probability of encountering more than 3 timeouts.

Would it be better to remove this retry and remove the stop condition on the other methods?

Reply:
I'd argue that if there's underlying instability in network conditions for the operating environment of your Kubernetes pods, you'd rather have an error than trying to force the API call to go through.

Reply:
@jingngson I understand your point of view, but let me describe de conditions that triggered me to create this PR in this way. We are running on AKS. The network conditions inside the cluster seem fine. However, the Kubernetes API (on the managed master nodes) has frequent timeouts. This results in worker pods that are hanging/waiting indefinitely on a response from the Kubernetes API. However, the underlying tasks that were launched (e.g. a Spark job) execute and finish succesfully. In this scenario, what we really want is the worker job to poll indefinitely for the status of launched tasks.

Perhaps the timeouts and stop conditions should be made configurable?